test/functional/neutronless/esd/test_esd.py

```diff
@@ -102,63 +104,127 @@ def esd():
+        icontrol_driver._common_service_handler(service)
+        validator.assert_virtual_deleted(listener, folder)
+
+        # delete loadbalancer
+        service = services['delete_loadbalancer']
+        icontrol_driver._common_service_handler(service, delete_partition=True)
+        assert not bigip.folder_exists(folder)
+    
+    request.addfinalizer(teardown)
+    apply_service, remove_service, esd_name = _set_esd(services, request)
+    ti = TESTINFRA(apply_service,
+                   remove_service,
+                   icontrol_driver,
+                   esd,
+                   validator,
+                   folder,
+                   listener,
+                   esd_name)
+    return ti
+
+def _set_esd(services, request):
+    literal_esd_name = request.function.__name__.partition('test_esd_')[2]
+    full_esd_name = 'f5_ESD_' + literal_esd_name
+    new_apply_esd = deepcopy(services["apply_ABSTRACT_ESD"])
+    new_remove_esd = deepcopy(services["remove_ABSTRACT_ESD"])
+    new_apply_esd['l7policies'][0]['name'] = full_esd_name
+    new_remove_esd['l7policies'][0]['name'] = full_esd_name
+    return new_apply_esd, new_remove_esd, full_esd_name
+    
+def _apply_validate_remove_validate(infra):
+    i = infra
     # apply ESD
-    service = service_iter.next()
-    icontrol_driver._common_service_handler(service)
-    validator.assert_esd_applied(esd['esd_demo_1'], listener, folder)
+    i.icontrol_driver._common_service_handler(i.apply_service)
+    i.validator.assert_esd_applied(i.esd[i.esd_name], i.listener, i.folder)
 
     # remove ESD
-    service = service_iter.next()
-    icontrol_driver._common_service_handler(service)
-    validator.assert_virtual_valid(listener, folder)
-    validator.assert_esd_removed(esd['esd_demo_1'], listener, folder)
+    i.icontrol_driver._common_service_handler(i.remove_service)
+    i.validator.assert_virtual_valid(i.listener, i.folder)
+    i.validator.assert_esd_removed(i.esd[i.esd_name], i.listener, i.folder)
 
-    # apply another ESD
-    service = service_iter.next()
-    icontrol_driver._common_service_handler(service)
-    validator.assert_esd_applied(esd['esd_demo_2'], listener, folder)
 
-    # remove ESD
-    service = service_iter.next()
-    icontrol_driver._common_service_handler(service)
-    validator.assert_virtual_valid(listener, folder)
-    validator.assert_esd_removed(esd['esd_demo_2'], listener, folder)
+def test_setup_teardown(setup):
+    pass
 
-    # delete pool (and member, node)
-    service = service_iter.next()
-    icontrol_driver._common_service_handler(service)
-    validator.assert_pool_deleted(pool, None, folder)
+def test_esd_two_irules(setup):
+    infrastructure = setup
+    _apply_validate_remove_validate(infrastructure)
 
-    # delete listener
-    service = service_iter.next()
-    icontrol_driver._common_service_handler(service)
-    validator.assert_virtual_deleted(listener, folder)
+def test_esd_lbaas_ctcp(setup):
```

Comment:
Generally add docstrings to indicate what the test is doing.

Reply:
You may add pep 257 check on the file.

Reply:
Good point.  I have added pep257 compliant docstrings.  I've also added module-specific pep257 in .travis.yml.